File: solutions/2026-08-25/java/2026-08-25-smallest-missing-multiple-of-k.java
```java
import java.util.HashSet;
import java.util.Set;

// Problem: Smallest Missing Multiple of K
// Link: https://leetcode.com/problems/smallest-missing-multiple-of-k/
// Approach:
// We need to find the smallest positive multiple of k that is not present in the given array nums.
// A straightforward approach is to iterate through positive multiples of k (k, 2k, 3k, ...)
// and for each multiple, check if it exists in the nums array. The first multiple of k
// that we don't find in nums is our answer.
// To efficiently check for the presence of numbers in nums, we can store all elements of nums
// in a HashSet. This allows for O(1) average time complexity for lookups.
//
// Time Complexity:
// - Inserting all elements of nums into a HashSet takes O(N) time, where N is the length of nums.
// - In the worst case, we might need to check up to approximately (max_value_in_nums / k) + 1 multiples of k.
//   Since nums[i] <= 100 and k >= 1, the maximum number of multiples of k we might check is bounded.
//   For example, if k=1, we might check up to 101 multiples. If k=100, we check at most 2 multiples.
//   The maximum number of multiples we check is approximately (100 / k) + 1.
//   So, the loop runs at most O(100/k) times.
// - Each check inside the loop (HashSet.contains) takes O(1) on average.
// - Therefore, the overall time complexity is O(N + 100/k). Given the constraints, N <= 100 and k >= 1,
//   this is effectively O(N) or O(100), which is considered constant time in terms of problem constraints.
//
// Space Complexity:
// - We use a HashSet to store the elements of nums. In the worst case, the HashSet will store all
//   N elements of nums.
// - Therefore, the space complexity is O(N), where N is the length of nums. Given N <= 100, this is O(100),
//   which is considered constant space in terms of problem constraints.

class Solution {
    /**
     * Finds the smallest positive multiple of k that is missing from the array nums.
     *
     * @param nums An integer array.
     * @param k    An integer.
     * @return The smallest positive multiple of k missing from nums.
     */
    public int smallestMissingMultipleOfK(int[] nums, int k) {
        // Use a HashSet for efficient O(1) average time lookups.
        Set<Integer> numSet = new HashSet<>();

        // Add all numbers from the input array to the HashSet.
        for (int num : nums) {
            numSet.add(num);
        }

        // Start checking multiples of k.
        // The first multiple is k itself (1 * k).
        // We increment the multiplier `i` (1, 2, 3, ...) to generate subsequent multiples.
        for (int i = 1; ; i++) {
            int multiple = i * k; // Calculate the current multiple of k.

            // Check if this multiple exists in the HashSet.
            if (!numSet.contains(multiple)) {
                // If the multiple is not found, it's the smallest missing multiple of k.
                return multiple;
            }
        }
    }
}
```

File: solutions/2026-08-25/python/2026-08-25-smallest-missing-multiple-of-k.py
```python
# Problem: Smallest Missing Multiple of K
# Summary: Find the smallest positive multiple of k that is not present in the given array nums.
# Link: https://leetcode.com/problems/smallest-missing-multiple-of-k/
#
# Approach:
# The problem asks for the smallest positive multiple of k that is missing from the array.
# Multiples of k are of the form m * k, where m is a positive integer (1, 2, 3, ...).
# We can iterate through positive multiples of k (k, 2k, 3k, ...) and check if each multiple
# is present in the input array `nums`. The first multiple of k that we do not find
# in `nums` is our answer.
#
# To efficiently check for the presence of numbers in `nums`, we can convert `nums` into
# a set. Set lookups (using the 'in' operator) have an average time complexity of O(1).
#
# We start checking from the first positive multiple of k (which is k itself, i.e., 1*k).
# Then we check 2*k, 3*k, and so on. The loop will continue until we find a multiple
# that is not in the set.
#
# Time Complexity:
# Converting `nums` to a set takes O(N) time, where N is the length of `nums`.
# In the worst case, we might have to check up to N+1 multiples of k if all the first N multiples
# are present in `nums`. If the smallest missing multiple is `M*k`, then we perform `M` lookups in the set.
# Since nums[i] <= 100 and k <= 100, the maximum possible value of a multiple we might need to check
# is related to the maximum value in nums and k. If all multiples up to N are present, the next multiple
# N+1 might be the answer. So M is roughly proportional to N.
# Thus, the overall time complexity is dominated by the set conversion and the loop, making it O(N) on average.
#
# Space Complexity:
# We use a set to store the elements of `nums` for efficient lookups. In the worst case,
# the set will store all N elements of `nums`. Therefore, the space complexity is O(N).

class Solution:
    def findSmallestMissingMultiple(self, nums: list[int], k: int) -> int:
        """
        Finds the smallest positive multiple of k that is missing from the input array.

        Args:
            nums: A list of integers.
            k: An integer representing the base for multiples.

        Returns:
            The smallest positive multiple of k not present in nums.
        """
        # Convert the input list to a set for efficient O(1) average time lookups.
        nums_set = set(nums)

        # Start checking multiples of k from 1*k, 2*k, 3*k, ...
        multiple_factor = 1
        while True:
            # Calculate the current multiple of k
            current_multiple = multiple_factor * k

            # Check if the current multiple is present in the set
            if current_multiple not in nums_set:
                # If it's not present, this is the smallest missing multiple.
                return current_multiple

            # If the current multiple is present, move to the next multiple.
            multiple_factor += 1

# Example usage (for testing purposes, not part of the LeetCode submission format):
# sol = Solution()
# print(sol.findSmallestMissingMultiple([8, 2, 3, 4, 6], 2))  # Output: 10
# print(sol.findSmallestMissingMultiple([1, 4, 7, 10, 15], 5)) # Output: 5
# print(sol.findSmallestMissingMultiple([1, 2, 3, 4, 5], 1))   # Output: 6
# print(sol.findSmallestMissingMultiple([10, 20, 30], 10)) # Output: 40
```